dipy/reconst/shm.py

```diff
@@ -240,6 +240,74 @@ def real_sph_harm(m, n, theta, phi):
     return real_sh
 
 
+def real_sym_sh_brainsuite(sh_order, theta, phi):
+    """
+    Compute the real spherical harmonics used in BrainSuite [1]
+
+    Parameters
+    ----------
+    sh_order : int
+        The maximum degree of the spherical harmonic basis.
+    theta : float [0, 2*pi]
+        The azimuthal (longitudinal) coordinate.
+    phi : float [0, pi]
+        The polar (colatitudinal) coordinate.
+
+    Returns
+    -------
+    y_mn : real float
+        The real harmonic $Y^m_n$ sampled at `theta` and `phi` as
+        implemented in BrainSuite.
+    m : array
+        The order of the harmonics.
+    n : array
+        The degree of the harmonics.
+
+    References
+    ----------
+    ..[1] Justin P. Haldar, Richard M. Leahy, "Linear transforms for Fourier
+          data on the sphere: Application to high angular resolution diffusion
+          MRI of the brain", NeuroImage, 2013.
+
+    """
+    def _legendre(n, X):
+        res = []
+        for m in range(n+1):
+            res.append(lpmv(m, n, X))
+        return np.row_stack(res)
+
+    # Original code used matlab conventions: convert dipy to matlab
+    theta, phi = phi, theta
+
+    # Adapted from BrainSuite's sph_harm_basis
+    ndirs = len(phi)
+    nbases = int(1 / 2 * (sh_order + 1) * (sh_order + 2))
+    S = np.zeros((ndirs, nbases))
+    S[:, 0] = 1 / np.sqrt(4 * np.pi)
+    L = np.zeros(nbases)
+    Z = np.zeros(nbases)
+    for ell in range(2, sh_order + 1, 2):
+        span = np.arange(ell + 1)
+        Pell = (np.ones((ndirs, ell + 1))
```

Comment:
rather compute the elements and then join then having such complicated formulation in a single line...